### PR TITLE
Sync with tj-ratelimiter

### LINKED ISSUE #12: Pull request - Add zremrangebyrank by penghap
[tj/node-ratelimiter]<br>
<br>
<b>Add zremrangebyrank</b><br>
By penghap <br>
<br>
Add zremrangebyrank to limit useful information count<br>
<br>
March 2, 2019 at 01:31PM<br>
via GitHub https://ift.tt/2Tr4yDk

### PR DESCRIPTION
**Pull request by 3imed-jaberi**
[@koa-ratelimiter](https://github.com/koajs/ratelimit) replace the [tj-ratelimiter](https://github.com/tj/node-ratelimiter) with `async-ratelimiter` ...

fix #12

### CODE REVIEW

**Comment by Kikobeats** [comment]
Thanks for this!

What do you think this change is needed? can you explain what it's doing?

**Comment by 3imed-jaberi** [comment]
- remove the options.decrease
- use the same operations array by add these to fileds:
  - ['zadd', key, now, now]
  - ['zremrangebyrank', key, 0, -(max + 1)]
- check for io-redis and support both redis modules (ioredis and node_redis)
- add resetMs to the returned object from the get method

**Comment by Kikobeats** [comment]
Need to remove `decrease` for docs as well

**Comment by Kikobeats** [comment]
I'm not interested into support `redis` module

**Comment by 3imed-jaberi** [comment]
> Need to remove `decrease` for docs as well

Done !

File: src/index.js [hunk 1]
```diff
@@ -33,24 +32,46 @@ module.exports = class Limiter {
...
+    const isIoRedis = Array.isArray(res[0])
+    const count = toNumber(isIoRedis ? res[1][1] : res[1])
+    const oldest = toNumber(isIoRedis ? res[3][1] : res[3])
+    const oldestInRange = toNumber(isIoRedis ? res[4][1] : res[4])
+    const resetMicro = (Number.isNaN(oldestInRange) ? oldest : oldestInRange) + duration * 1000
 
     return {
       remaining: count < max ? max - count : 0,
       reset: Math.floor(resetMicro / 1000000),
+      resetMs: Math.floor(resetMicro / 1000),
```
Comment (Kikobeats) [inline src/index.js:52]: Can you remove this?

Prefer to keep output simple; you can always calculate `resetMs` if you need in your side

**Review by Kikobeats — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -97,9 +96,11 @@ You can pass this value using when you use `.get` method as well.
...
 
 Given an `id`, returns a Promise with the status of the limit with the following structure:
-  - `total`: `max` value.
-  - `remaining`: number of calls left in current `duration` without decreasing current `get`.
-  - `reset`: time since epoch in seconds that the rate limiting period will end (or already ended).
+
+- `total`: `max` value.
+- `remaining`: number of calls left in current `duration` without decreasing current `get`.
+- `reset`: time since epoch in seconds that the rate limiting period will end (or already ended).
+- `resetMs`: time since epoch in milliseconds that the rate limiting period will end (or already ended).
```
Comment (Kikobeats) [inline README.md:103]:
```suggestion
```

File: src/index.js [hunk 3]
```diff
@@ -33,24 +32,46 @@ module.exports = class Limiter {
...
-      (Number.isNaN(oldestInRange) ? oldest : oldestInRange) + duration * 1000
+    const isIoRedis = Array.isArray(res[0])
+    const count = toNumber(isIoRedis ? res[1][1] : res[1])
+    const oldest = toNumber(isIoRedis ? res[3][1] : res[3])
+    const oldestInRange = toNumber(isIoRedis ? res[4][1] : res[4])
+    const resetMicro = (Number.isNaN(oldestInRange) ? oldest : oldestInRange) + duration * 1000
 
     return {
       remaining: count < max ? max - count : 0,
       reset: Math.floor(resetMicro / 1000000),
```
Comment (Kikobeats) [inline src/index.js:51]:
```suggestion
      reset: Math.floor(resetMicro / 1000000)
```

**Review by Kikobeats — CHANGES_REQUESTED** [review]

**Comment by Kikobeats** [comment]
Thanks a lot!

**Pull request merged**